features/0746-shorten-url/README.md

```diff
@@ -0,0 +1,167 @@
+# Aries RFC 0746: Shorten URL Protocol 1.0
+
+- Authors: [Timo Glastra](mailto:timo@animo.id) (Animo Solutions)
+- Status: [PROPOSED](/README.md#proposed)
+- Since: 2022-07-26
+- Status Note: Initial version
+- Start Date: 2022-07-20
+- Tags: [feature](/tags.md#feature), [protocol](/tags.md#protocol)
+
+## Summary
+
+A protocol to request a shortened URL for a given URL.
+
+## Motivation
+
+URL shorteners are common technology used by a variety of services. There's a lot of free to use url shorteners available, but those bring in privacy concerns and don't always support invalidating the URL. When an agent can't host shortened url themselves (e.g. in the case of a mobile edge agent), having another agent host the shortened url provides a good solution to dynamically shorten the url. This would allow agents to create QR codes of invitations that are easy to scan in size, and also allows to share the shortened with other agents out of band
+
+> Note: Some platforms such as iOS remove the query from an url when you click on it if the query is too long. This is problematic for out of band invitations that rely on the `oob` property to be present
+> in the invitation url.
+> E.g. in the case of the following url: `https://my-url.com?oob=1234`, the url when clicked on will be interpreted as `https://my-url.com`.
+
+## Tutorial
+
+### Name and Version
+
+URI: `https://didcomm.org/shorten-url/1.0`
+
+Protocol Identifier: `shorten-url`
+
+Version: `1.0`
+
+### Roles
+
+**url-shortener**
+
+**long-url-provider**
+
+The **url-shortener** is an agent who will provide the **long-url-provider** with a shortened url. The **long-url-provider** can then share this shortened link. The **url-shortener** will return the long url when the short url is fetched.
+
+### Messages
+
+#### Request Shortened URL
+
+Message sent by the **long-url-provider** to the **url-shortener** to request a shortened version of an url.
+
+```json
+{
+  "@type": "https://didcomm.org/shorten-url/1.0/request-shortened-url",
+  "@id": "<UUID>",
+  "url": "<URL>",
+  "requested_validity_seconds": <INTEGER>,
+  "goal_code": "<GOAL_CODE>"
```

Comment:
Perhaps add a vanity or beautify string here? For example, I would like to request....

example.org/my-brand-connect

Something like

"vanity": "<URL-SAFE-STRING>"

"vanity": "my-brand-connect"

Reply:
This seems to be a feature that is available in most url shorteners. Adding it!